src/openrct2/interface/Screenshot.cpp

```diff
@@ -207,106 +212,40 @@ std::string screenshot_dump_png_32bpp(int32_t width, int32_t height, const void*
-                case 2:
-                    return { lower, upperY };
-                case 3:
-                    return { lower, lower };
-            }
-        case EdgeType::TOP:
-            switch (rotation)
-            {
-                default:
-                case 0:
-                    return { lower, lower };
-                case 1:
-                    return { upperX, lower };
-                case 2:
-                    return { upperX, upperY };
-                case 3:
-                    return { lower, upperY };
-            }
-        case EdgeType::RIGHT:
-            switch (rotation)
-            {
-                default:
-                case 0:
-                    return { lower, upperY };
-                case 1:
-                    return { lower, lower };
-                case 2:
-                    return { upperX, lower };
-                case 3:
-                    return { upperX, upperY };
-            }
-        case EdgeType::BOTTOM:
-            switch (rotation)
-            {
-                default:
-                case 0:
-                    return { upperX, upperY };
-                case 1:
-                    return { lower, upperY };
-                case 2:
-                    return { lower, lower };
-                case 3:
-                    return { upperY, lower };
-            }
-    }
-}
-
-static int32_t GetHighestBaseClearanceZ(const CoordsXY& location)
+static int32_t GetHighestBaseClearanceZ(const CoordsXY& location, const bool useViewClipping)
 {
     int32_t z = 0;
     auto element = map_get_first_element_at(location);
     if (element != nullptr)
     {
         do
         {
+            if (useViewClipping && (element->GetBaseZ() > gClipHeight * COORDS_Z_STEP))
+            {
+                continue;
+            }
             z = std::max<int32_t>(z, element->GetBaseZ());
             z = std::max<int32_t>(z, element->GetClearanceZ());
         } while (!(element++)->IsLastForTile());
     }
     return z;
 }
 
-static int32_t GetTallestVisibleTileTop(const TileCoordsXY& mapSize, int32_t rotation)
+static int32_t GetTallestVisibleTileTop(
+    int32_t rotation, TileCoordsXY startCoords, TileCoordsXY endCoords, const bool useViewClipping)
 {
-    int32_t minViewY = 0;
-    for (int32_t y = 1; y < mapSize.y - 1; y++)
+    int32_t minViewY = INT32_MAX;
+    for (int32_t y = startCoords.y; y <= endCoords.y; y++)
     {
-        for (int32_t x = 1; x < mapSize.x - 1; x++)
+        for (int32_t x = startCoords.x; x <= endCoords.x; x++)
         {
             auto location = TileCoordsXY(x, y).ToCoordsXY();
-            int32_t z = GetHighestBaseClearanceZ(location);
+            int32_t z = GetHighestBaseClearanceZ(location, useViewClipping);
             int32_t viewY = translate_3d_to_2d_with_z(rotation, CoordsXYZ(location, z)).y;
             minViewY = std::min(minViewY, viewY);
         }
     }
-    return minViewY - 256;
+    return minViewY - 32;
```

Comment:
This is for some padding at the top?

Reply:
Currently, the padding is only required because of the weird collision check of the observation tower at the top. Everything else seems to work as it should, which means that the sprite will not be cropped off.

Reply:
to be more precise, the bounding box on the observation tower at the top appears to be lower than the actual sprite and it is possible to place things on top of the observation tower (which probably was always the case)

Reply:
Yeah, there are quite some objects, track elements, shops, and entrance styles in the game where the clearance height does not match the visuals. I guess the original 256 pixels was for this, I wonder if 32 is enough. 256 seems a bit too much, though.

Any way, I'd like this commented, to explain why the magic number is there.

Reply:
Comment was added

Reply:
Nitpicky, but we usually place comments above code, not behind it. Comments are only placed behind a line for lists and other situations where a comment would interrupt the code.

Reply:
Fixed comment placement